### PR TITLE
Add blob operations in restart_socksql test

### PR DESCRIPTION
**Pull request by riverszhang89**
Updating a blob consists of a couple of OSQL opcodes - OSQL_USEDB, OSQL_UPDCOLS, OSQL_QBLOB, OSQL_UPDREC and finally OSQL_DONE. If any of them fails, we should retry the operation from the beginning. The original OSQL retry logic did not handle OSQL_QBLOB failure correctly. Therefore, a restarted OSQL session would miss OSQL_QBLOB, resulting in the "no blobs flags with blob buffers" error.

The bug was fixed in 46f9eb3856d3b925825e41dd5988233441fbfda1. However we don't have a testcase for it. The commit adds blob updates to restart_socksql test.

### CODE REVIEW

File: db/osqlsqlthr.c [hunk 1]
```diff
@@ -144,15 +144,15 @@ static inline int osql_should_restart(struct sqlclntstate *clnt, int rc)
...
         if (rc) {                                                              \
             logmsg(LOGMSG_ERROR,                                               \
                    "%s: failed to restart socksql session rc=%d\n", __func__,  \
                    rc);                                                        \
+        } else {                                                               \
+            restarted = 1;                                                     \
         }                                                                      \
-    }                                                                          \
-    if (rc) {                                                                  \
+    } else if (rc) {                                                           \
```
Comment (ldennis) [inline db/osqlsqlthr.c]: if `osql_sock_restart` fails, do we want to coalesce rcode to `SQLITE_INTERNAL`?

**Comment by ldennis** [comment]
Looks good otherwise.

File: db/osqlsqlthr.c (hunk 1 above)
Comment (riverszhang89) [reply db/osqlsqlthr.c]: Good point. Will undo this change.

**Comment by riverszhang89** [comment]
@ldennis Appreciate the review!

**Pull request merged**